Framework/Algorithms/src/EQSANSCorrectFrame.cpp

```diff
@@ -58,32 +65,41 @@ void EQSANSCorrectFrame::exec() {
   const double minTOF = getProperty("MinTOF");
   const double frameWidth = getProperty("FrameWidth");
   const bool isFrameSkipping = getProperty("FrameSkipping");
-  const auto &spectrumInfo = inputWS->spectrumInfo();
+  const bool pathToPixel = getProperty("PathToPixel");
+  const std::string detectorName = getProperty("DetectorName");
 
+  const auto &spectrumInfo = inputWS->spectrumInfo();
+  auto ins = inputWS->getInstrument();
+  auto sam = ins->getSample();
+  auto mod = ins->getSource();
+  const auto msd = mod->getDistance(*sam);
```

Comment:
This could be done with [`spectrumInfo.l1()`](https://docs.mantidproject.org/nightly/api/python/mantid/api/SpectrumInfo.html#mantid.api.SpectrumInfo.l1)

Reply:
done